### ISSUE TITLE
Lookup Table API Enhancements: Allow single row updates

### ISSUE THREAD

**Issue by sethdarragile6** [CDCgov/prime-reportstream#5706]
## Problem statement

Currently in the Lookup Table API, in order to update even a single row in the table, you must submit the _entire content_ of the updated lookup table. While this has thus far met the needs for OnO, when using this from a UI, the experience is inelegant and expensive- requiring in one POST the sending of 1600+ element JSON. We would like a way to submit _one_ row for update.

In the [design doc](https://cdc-my.sharepoint.com/:w:/r/personal/srz7_cdc_gov/_layouts/15/doc2.aspx?sourcedoc=%7B107af8c4-c9be-47ad-8d6a-acc2817ae24b%7D&action=editnew&wdPreviousSession=88175852-f506-4502-9155-747b9f56bf2e&wdNewAndOpenCt=1652735315621&wdo=4&wdOrigin=wacFileNew&wdTpl=blank&wdLcid=1033&wdPreviousCorrelation=164fe638-ea15-4dca-a1dd-819218e6beaf&cid=2d68a5ae-8cbf-40f5-844a-d11607d174c0) for Valueset Editor UI, we outline two possible approaches- a "short lift" and a "heavy lift":

> “short lift”: if the API still kept its “insert full table at once” approach, row level posts could still be supported if key information about the row was posted with it, by... 
> - Querying the full set of rows by the key information (minus the row being replaced), then 
> - Writing the full set of rows back (plus the new replacement row) with a new version # 
> 
> “heavy lift”: API keeps full table copies, however it would do it in a “full + incremental(s)” way. In other words, a brand new lookup table would be fully inserted but any inserts, updates or deletes that came after that would simply track as deltas from the first full set. This is obviously a major refactor of the API, but would have the added advantage of saving a great deal of disk space all around 

## What you need to know

- [Valueset Editor Backend Architecture](https://cdc-my.sharepoint.com/:w:/r/personal/srz7_cdc_gov/_layouts/15/doc2.aspx?sourcedoc=%7B107af8c4-c9be-47ad-8d6a-acc2817ae24b%7D&action=editnew&wdPreviousSession=88175852-f506-4502-9155-747b9f56bf2e&wdNewAndOpenCt=1652735315621&wdo=4&wdOrigin=wacFileNew&wdTpl=blank&wdLcid=1033&wdPreviousCorrelation=164fe638-ea15-4dca-a1dd-819218e6beaf&cid=2d68a5ae-8cbf-40f5-844a-d11607d174c0)
- Related epic: #3489 
- Related tickets: 
-- #5534 
-- #5535  
-- #5590 

## Acceptance criteria

- [ ] Create a new endpoint (or extend an existing endpoint) to update a single lookup table row, instead of the entire table
- [ ] An update to a single table row must generate a new version of a table while keeping the old version.
- [ ] Unit tests as needed

**Comment by JGreenFearless** [comment]
@sethdarragile6 Is this something that requires work from the Pipeline Team?

**Comment by sethdarragile6** [comment]
@JGreenFearless not necessarily. it's backend/API work that traditionally would be done by the pipeline team, but also doable by experience (w/pipeline's eyes/help) @bgantick
